src/lucky/route_handler.cr

```diff
@@ -1,15 +1,36 @@
-require "colorize"
-
 class Lucky::RouteHandler
   include HTTP::Handler
 
+  Habitat.create do
+    setting mime_extensions : Hash(String, String) = {} of String => String
+  end
+
   def call(context)
+    original_path = context.request.path
+    format = get_extension(context.request.path)
+    context.request.path = context.request.path.gsub(format.to_s, "") if format
     handler = Lucky::Router.find_action(context.request)
     if handler
+      if format && (content_type = MIME.from_extension?(format))
+        Lucky.logger.debug({path_extension: format, new_content_type: content_type})
```

Comment:
Ah nice! Maybe we make it a warning so it is yellow?

Reply:
And maybe include “old_content_type” ? So it’s clear it is changing and how?

Reply:
Or maybe just “content_type_changed_to” a bit longer but I think clearer that it has been changed because of the extension